pkg/util/qos/oom_priority.go

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright 2022 The Katalyst Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package qos
+
+import (
+	apiconsts "github.com/kubewharf/katalyst-api/pkg/consts"
+)
+
+const (
+	IgnoreOOMPriorityScore         int = -300
+	ReclaimedCoresOOMPriorityScore int = -100
+	SharedCoresOOMPriorityScore    int = 0
+	DedicatedCoresOOMPriorityScore int = 100
+	SystemCoresOOMPriorityScore    int = 200
+	TopOOMPriorityScore            int = 300
+)
+
+func AlignOOMPriority(qosLevel string, userSpecifiedScore *int) (oomPriorityScore int) {
+	// Adjust OOM priority score if user specified
+	if userSpecifiedScore != nil {
+		oomPriorityScore = *userSpecifiedScore
+		// Adjust OOM Priority if it's out of range
```

Comment:
if oomPriorityScore == TopOOMPriorityScore {
   return oomPriorityScore
}

no need to check evey time below?